kafka-rest/src/main/java/io/confluent/kafkarest/response/CrnFactoryImpl.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.response;
+
+import io.confluent.kafkarest.config.ConfigModule.CrnAuthorityConfig;
+import java.util.StringJoiner;
+import javax.inject.Inject;
+
+/**
+ * A {@link CrnFactory} that uses the given {@code crnAuthorityConfig} as the Confluent Resource
+ * Authority for the generated CRNs.
+ */
+public final class CrnFactoryImpl implements CrnFactory {
```

Comment:
can kafka-rest not use the `CRN` class that @WanderingStar added to ce-kafka? (i don't know how the dependencies flow here)

Reply:
I think this rest-proxy code has minimal dependencies, and specifically not `ce-kafka`, which makes sense. `ce-kafka` is huge. But the CRN code is isolated enough to just move the module out of `ce-kafka` cleanly.

Reply:
ce-kafka is closed source, right? so we can't. also i would say having an extra dependency on account of just this isn't worth it.

Reply:
As @mhowlett pointed out, we can't depend in ce-kafka. We should extract the CRN helpers out in a public repo. For the time being, since this is going to 5.5, I will replicate functionality. To be revisited for 6.0.